sklearn/neighbors/_classification.py

```diff
@@ -437,7 +440,6 @@ def __init__(
         outlier_label=None,
         metric_params=None,
         n_jobs=None,
-        **kwargs,
```

Comment:
I just spoke with @ogrisel IRL, and it seems better to indeed deprecate `kwargs`. So we still need an entry in what's new but this time it should announce the deprecation instead of the hard removal.

So what we need to do here is to check the length of `kwargs` in `__init__` and raise a `FutureWarning` mentioning that the parameters are ignored and that in 1.2 (or 1.3 since we might merge after the release) an error will be raised instead of this warning.

We will also need a small test to check that we properly raise the warning.